front/app/components/Form/Components/Controls/visibilityUtils.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ Reference for some of the code in this file: https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/testers/testers.ts
+ * 
+*/
+
+import Ajv from 'ajv';
+import {
+  UISchemaElement,
+  Rule,
+  Condition,
+  RuleEffect,
+  OrCondition,
+  AndCondition,
+  LeafCondition,
+  SchemaBasedCondition,
+  Scopable,
+  composeWithUi,
+  resolveData,
+} from '@jsonforms/core';
+import { has } from 'lodash-es';
+import { PageType } from '../Layouts/utils';
+
+export type ExtendedRule = {
+  /**
+   * The effect of the rule
+   */
+  effect: RuleEffect;
+  /**
+   * The condition of the rule that must evaluate to true in order
+   * to trigger the effect.
+   */
+  condition: Condition;
+} & Rule;
+
+export type ExtendedUISchema = {
+  ruleArray?: ExtendedRule[];
+  label?: string;
+} & UISchemaElement &
+  Scopable;
+
+const isOrCondition = (condition: Condition): condition is OrCondition =>
+  condition.type === 'OR';
+
+const isAndCondition = (condition: Condition): condition is AndCondition =>
+  condition.type === 'AND';
+
+const isLeafCondition = (condition: Condition): condition is LeafCondition =>
+  condition.type === 'LEAF';
+
+const isSchemaCondition = (
+  condition: Condition
+): condition is SchemaBasedCondition => has(condition, 'schema');
+
+const getConditionScope = (condition: Scopable, path: string): string => {
+  return composeWithUi(condition, path);
+};
+
+const evaluateCondition = (
+  data: any,
+  condition: Condition,
+  path: string,
+  ajv: Ajv
+): boolean => {
+  if (isAndCondition(condition)) {
+    return condition.conditions.reduce(
+      (acc, cur) => acc && evaluateCondition(data, cur, path, ajv),
+      true
+    );
+  } else if (isOrCondition(condition)) {
+    return condition.conditions.reduce(
+      (acc, cur) => acc || evaluateCondition(data, cur, path, ajv),
+      false
+    );
+  } else if (isLeafCondition(condition)) {
+    const value = resolveData(data, getConditionScope(condition, path));
+    return value === condition.expectedValue;
```

Comment:
We are not using them but I didn't want to limit the initial API because technically those are still in their implementation. I'm reusing what they have so the person reusing our logic will have extended capabilities without necessarily limiting what the library provides

https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/util/runtime.ts#L61